config/plugins/sonic-pbh_yang.py

```diff
@@ -0,0 +1,747 @@
+            option_name: name of 'click' option
+    """
+
+    if _list is None:
+        return
+
+    table = db.cfgdb.get_table(conf_db_key)
+    correct_list = list(table.keys())
+
+    splited_list = _list.split(',')
+
+    for elem in splited_list:
+        if elem not in correct_list:
+            exit_with_error("Error: invalid value '{}' for '{}' option, please use {}".format(elem, option_name, correct_list), fg="red")
+
+
+def ip_mask_hash_field_correspondence(ip_mask, hash_field):
+    # at this point
+    # --ip_mask value can be None or VALID IP
+    # --hash_field always valid
+
+    if ((ip_mask is None) and
+        (hash_field == 'INNER_DST_IPV4' or hash_field == 'INNER_SRC_IPV4' or
+        hash_field == 'INNER_DST_IPV6' or hash_field == 'INNER_SRC_IPV6')):
+        exit_with_error("Error: the value of --hash-field='{}' is NOT compatible with the value of --ip-mask='{}'".format(hash_field, ip_mask), fg='red')
+
+    if ((ip_mask is not None) and
+        (hash_field == 'INNER_IP_PROTOCOL' or
+        hash_field == 'INNER_L4_DST_PORT' or
+        hash_field == 'INNER_L4_SRC_PORT')):
+        exit_with_error("Error: the value of --hash-field='{}' is NOT compatible with the value of --ip-mask='{}'".format(hash_field, ip_mask), fg='red')
+
+    if ((hash_field == 'INNER_DST_IPV4' or hash_field == 'INNER_SRC_IPV4') and
+        (not isinstance(ipaddress.ip_address(ip_mask), IPv4Address))):
+        exit_with_error("Error: the value of --hash-field='{}' is NOT compatible with the value of --ip-mask='{}'".format(hash_field, ip_mask), fg='red')
+
+    if ((hash_field == 'INNER_DST_IPV6' or hash_field == 'INNER_SRC_IPV6') and
+        (not isinstance(ipaddress.ip_address(ip_mask), IPv6Address))):
+        exit_with_error("Error: the value of --hash-field='{}' is NOT compatible with the value of --ip-mask='{}'".format(hash_field, ip_mask), fg='red')
+
+def update_ip_mask_hash_field(db, hash_field_name, ip_mask, hash_field):
+
+    if (ip_mask is None) and (hash_field is None):
+        return
+
+    table = db.cfgdb.get_table('PBH_HASH_FIELD')
+    hash_field_obj = table[hash_field_name]
+
+    if (ip_mask is None) and (hash_field is not None):
+
+        try:
+            ip_mask = hash_field_obj['ip_mask']
+        except Exception as e:
+            ip_mask = None
+
+        ip_mask_hash_field_correspondence(ip_mask, hash_field)
+
+    if (ip_mask is not None) and (hash_field is None):
+
+        try:
+            hash_field = hash_field_obj['hash_field']
+        except Exception as e:
+            hash_field = None
+
+        ip_mask_hash_field_correspondence(ip_mask, hash_field_obj['hash_field'])
+
+
+@click.group(name='pbh',
+             cls=clicommon.AliasedGroup)
+def PBH():
+    """ Configure PBH (Policy based hashing) feature """
+
+    pass
+
+
+@PBH.group(name="hash-field",
+           cls=clicommon.AliasedGroup)
+def PBH_HASH_FIELD():
+    """ Configure PBH hash field """
+
```

Comment:
@vadymhlushko-mlnx generic comment: what's the purpose of this empty line? Is it by PEP8?

Reply:
pep8